ReactWindows/ReactNative/UIManager/UIViewOperationQueueCollection.cs

```diff
@@ -0,0 +1,564 @@
+                ReactChoreographer reactChoreographer = ReactChoreographer.CreateSecondaryInstance(foundView);
+
+                queueInfo = new QueueInstanceInfo()
+                {
+                    queueInstance = new UIViewOperationQueueInstance(
+                        _reactContext,
+                        new NativeViewHierarchyManager(_viewManagerRegistry, rootViewDispatcher, OnViewsDropped),
+                        reactChoreographer),
+                    rootViewCount = 1
+                };
+
+                // Add new tuple to map
+                _dispatcherToOperationQueueInfo.Add(rootViewDispatcher, queueInfo);
+
+                if (_active)
+                {
+                    // Simulate an OnResume from the correct dispatcher thread
+                    // (OnResume/OnSuspend/OnDestroy have this thread affinity, all other methods do enqueuings in a thread safe manner)
+                    DispatcherHelpers.RunOnDispatcher(rootViewDispatcher, queueInfo.queueInstance.OnResume, true); // inlining allowed
+                }
+            }
+            else
+            {
+                // Queue instance does exist.
+                // Increment the count of root views. This is helpful for the case the count reaches 0 so we can cleanup the queue.
+                queueInfo.rootViewCount++;
+            }
+
+            // Add tag
+            _reactTagToOperationQueue.Add(tag, queueInfo.queueInstance);
+
+            // Send forward
+            queueInfo.queueInstance.AddRootView(tag, rootView, themedRootContext);
+       }
+
+        /// <summary>
+        /// Enqueues an operation to remove the root view.
+        /// </summary>
+        /// <param name="rootViewTag">The root view tag.</param>
+        public void EnqueueRemoveRootView(int rootViewTag)
+        {
+            // Called on layout manager thread
+
+            UIViewOperationQueueInstance queue = GetQueueByTag(rootViewTag);
+
+            // Send forward
+            queue.EnqueueRemoveRootView(rootViewTag);
+
+            // Do some maintenance/cleanup if needed.
+            // Find the queue info
+            var pair = _dispatcherToOperationQueueInfo.First(p => p.Value.queueInstance == queue);
+
+            // Decrement number of root views
+            pair.Value.rootViewCount--;
+
+            if (queue != MainUIViewOperationQueue)
+            {
+                if (pair.Value.rootViewCount == 0)
+                {
+                    // We can remove this queue and then destroy
+                    _dispatcherToOperationQueueInfo.Remove(pair.Key);
+
+                    // Simulate an OnDestroy from the correct dispatcher thread
+                    // (OnResume/OnSuspend/OnDestroy have this thread affinity, all other methods do enqueuings in a thread safe manner)
+                    DispatcherHelpers.RunOnDispatcher(pair.Key, queue.OnDestroy);
+                }
```

Comment:
Why is inlining no longer allowed here?

Reply:
I thought it would be confusing, the code executes in layout manager thread, so there's no chance for inlining anyway (well, except the debug code or that special case where LM thread couldn't be created. I didn't want to change flows thore those cases).

---
In reply to: [167538747](https://github.com/Microsoft/react-native-windows/pull/1615#discussion_r167538747) [](ancestors = 167538747)